### ISSUE TITLE
[Bug]: SPM Package download fails to verify checksum

### ISSUE THREAD

**Issue by lemonandlime** [OneSignal/OneSignal-iOS-SDK#1144]
### What happened?

When adding the package with SPM to a project the checksum of of the package or binary can not be verified and Xcode raises a fatal error. This happens booth in the gui add step when you add in Xcode menus and when you add it to another SPM package as a dependency.

<img width="1114" alt="Screenshot 2022-10-28 at 10 33 31" src="https://user-images.githubusercontent.com/2911240/198543503-d45f7fe2-55ab-4999-9768-2d8ba3b8dd26.png">

### Steps to reproduce?

```Markdown
1. Create new xcode project 
2. Add OneSignal as a SPM dependency
```

### What did you expect to happen?

The SPM package should be downloaded and added to the project without errors

### OneSignal iOS SDK version

3.11.4

### iOS version

15

### Specific iOS version

```Markdown
iOS version does not apply here but my system setup is:
- macos ventura 13.0
- xcode 14.0.1
- Specified version of OneSignal SPM `UpToNextMajor 3.0.0`
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by SamuelKebis** [comment]
Similarly, here.

<img width="1023" alt="Screenshot 2022-10-28 at 10 09 45" src="https://user-images.githubusercontent.com/112950762/198546671-400d7f83-885a-426e-a9b9-b00ecefe4426.png">

I wanted to contribute on project that is using this Package and I cannot build without this checksum error.

I’ve already tried (and the error persist):
- Clean build folder
- Reset Package Cache
- Resolve Package Versions
- Update To Latest Versions
- Deleted DerivedData
- Deleted caches from Library/Caches:
   - com.apple.dt.Xcode
   - com.apple.dt.Xcode.sourcecontrol.Git
   - com.apple.dt.xcodebuild
   - com.github.GitHubClient
   - com.github.GitHubClient.ShipIt
   - org.swift.swiftpm
- Deleted the whole project and fetched it again
- Restarted Xcode
- Restarted mac

```
macOS: Monterey 12.6
Xcode version: 14.0.1
```

**Comment by dams229** [comment]
Same issue with XCode 14.1 RC 2 here.
Release `3.11.4` has incorrect checksum: 
https://github.com/OneSignal/OneSignal-iOS-SDK/commit/40bcc86eb5fc4bcf0e54d311840ecc8533f589b6#diff-f913940c58e8744a2af1c68b909bb6383e49007e6c5a12fb03104a9006ae677e

"OneSignal" binary target checksum should be `873600d3a903929baae530f059b47456250573089b044d48f771131083cd4dcb`, same for other targets:

```
  checksum of downloaded artifact of binary target 'OneSignalExtension' (d8930e1a48e67c0d3d8a4dd4b7348c74459012736ba5f866ea90aa444504ef27) does not match checksum specified by the manifest (994a897b849dacc873c22e5b0e2db9457e736c9b721ae6d0aaa1849130961e9a)
  checksum of downloaded artifact of binary target 'OneSignalOutcomes' (e347fbfd2810f390e871df33b3255a9a282d157bbf4c03b3bcbcd14b1e7fdae2) does not match checksum specified by the manifest (65b57954ce3c16019a7562d8e3e180c987e1aab04406a7c2c5825cb92a373678)
  checksum of downloaded artifact of binary target 'OneSignalCore' (b5c579fe86e11ab5916a964cc83f6181102cb4b91fba97cb3aeddb772347c9c0) does not match checksum specified by the manifest (be0d575b6f5ad50e59e7fe49a0b808fe4cc2833358620dd7b177fe5ff7b47dea)
  checksum of downloaded artifact of binary target 'OneSignal' (873600d3a903929baae530f059b47456250573089b044d48f771131083cd4dcb) does not match checksum specified by the manifest (ff0c6f6c793c5d1a515f516d7f7a624e1f0ba0e5ec32bc8572e352363ead06d6)
```

Forcing OneSignal dependency to 3.11.3 (`exactVersion`) fixed the issue for us.

**Comment by jkasten2** [comment]
@lemonandlime @SamuelKebis @dams229 Thanks for reporting. We just merged the updated checksums into `main` and retagged our release to resolve this issue when using `https://github.com/OneSignal/OneSignal-iOS-SDK` for URL for SwiftPM.


Using `https://github.com/OneSignal/OneSignal-XCFramework` is recommend instead as this is a much smaller download size, as Xcode clones the full git history which is quite large on `OneSignal-iOS-SDK`. However either URL should work and can confirm both checksums match now.

**Comment by guidev** [comment]
The same is happening with `https://github.com/OneSignal/OneSignal-XCFramework` 3.11.4

```
checksum of downloaded artifact of binary target 'OneSignalCore' (b5c579fe86e11ab5916a964cc83f6181102cb4b91fba97cb3aeddb772347c9c0) does not match checksum specified by the manifest (be0d575b6f5ad50e59e7fe49a0b808fe4cc2833358620dd7b177fe5ff7b47dea)
checksum of downloaded artifact of binary target 'OneSignalOutcomes' (e347fbfd2810f390e871df33b3255a9a282d157bbf4c03b3bcbcd14b1e7fdae2) does not match checksum specified by the manifest (65b57954ce3c16019a7562d8e3e180c987e1aab04406a7c2c5825cb92a373678)
checksum of downloaded artifact of binary target 'OneSignalExtension' (d8930e1a48e67c0d3d8a4dd4b7348c74459012736ba5f866ea90aa444504ef27) does not match checksum specified by the manifest (994a897b849dacc873c22e5b0e2db9457e736c9b721ae6d0aaa1849130961e9a)
checksum of downloaded artifact of binary target 'OneSignal' (873600d3a903929baae530f059b47456250573089b044d48f771131083cd4dcb) does not match checksum specified by the manifest (ff0c6f6c793c5d1a515f516d7f7a624e1f0ba0e5ec32bc8572e352363ead06d6)
```

**Comment by nan-li** [comment]
It looks like after we re-released `3.11.4`, the checksums that are being referred to here are the checksums from the previous release's Package.swift, while using the new uploaded packages. For example:

> checksum of downloaded artifact of binary target 'OneSignal' (873600d3a903929baae530f059b47456250573089b044d48f771131083cd4dcb) does not match checksum specified by the manifest (ff0c6f6c793c5d1a515f516d7f7a624e1f0ba0e5ec32bc8572e352363ead06d6)

`8736` is in the [latest Package.swift ](https://github.com/OneSignal/OneSignal-iOS-SDK/pull/1142/files) and `ff0c` is from the [last Package.swift](https://github.com/OneSignal/OneSignal-iOS-SDK/pull/1139/commits/b0999bdc9206d6224cf600483cbf0aff9c3c6440). 

We will have to fix.

**Comment by fhboswell** [comment]
For now try the OneSignal-iOS-SDK swift package instead of the OneSignal-XCFramework package. We are planning a new version shortly. 

We made a change under the same tag and we are having a caching issue with SPM.
